dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/task/SubTaskProcessor.java

```diff
@@ -129,14 +129,29 @@ private void updateTaskState() {
                 subProcessInstance.getId(),
                 subProcessInstance.getState());
         if (subProcessInstance != null && subProcessInstance.getState().isFinished()) {
-            // todo: check the status and transform
-            taskInstance.setState(TaskExecutionStatus.of(subProcessInstance.getState().getCode()));
+            taskInstance.setState(subProcessFinishedStateTransform(subProcessInstance.getState()));
             taskInstance.setEndTime(new Date());
             dealFinish();
             processService.saveTaskInstance(taskInstance);
         }
     }
 
+    private TaskExecutionStatus subProcessFinishedStateTransform(WorkflowExecutionStatus state) {
+        if (state.isSuccess()) {
+            return TaskExecutionStatus.SUCCESS;
+        } else if (state.isFailure()) {
+            return TaskExecutionStatus.FAILURE;
+        } else if (state.isStop()) {
+            return TaskExecutionStatus.KILL;
```

Comment:
why we workflow stop is kill?

Reply:
Stop is workflow state, and kill is task state. AFAK, task has no STOP state (`TaskExecutionStatus` has STOP state, but it has not been used anywhere). `TaskExecutionStatus` has code:

```
    public boolean isFinished() {
        return isSuccess() || isKill() || isFailure() || isPause();
    }
```

Pause workflow has pause task, only kill task state can match stop workflow state.